cluster/src/main/java/com/hartwig/pipeline/PipelineMain.java

```diff
@@ -227,16 +237,18 @@ private String collectTurquoiseSubject(final SomaticRunMetadata somaticRunMetada
     }
 
     @NotNull
-    private OutputPublisher getOutputPublisher(final Arguments arguments, final Storage storage) {
+    private OutputPublisher getOutputPublisher(final Arguments arguments, final Storage storage) throws IOException {
         if (arguments.context().equals(Pipeline.Context.PLATINUM)) {
             return new NoopOutputPublisher();
         } else {
-            return createPublisher(new EventPublisher<>(arguments.pubsubProject().orElse(arguments.project()),
-                    EventContext.builder()
-                            .environment(arguments.pubsubTopicEnvironment().orElseThrow(PipelineMain::missingPubsubArgumentsException))
-                            .workflow(arguments.pubsubTopicWorkflow().orElseThrow(PipelineMain::missingPubsubArgumentsException))
-                            .build(),
-                    new PipelineComplete.EventDescriptor()), arguments.context(), storage, arguments);
+            var eventContext = EventContext.builder()
+                    .environment(arguments.pubsubTopicEnvironment().orElseThrow(PipelineMain::missingPubsubArgumentsException))
+                    .workflow(arguments.pubsubTopicWorkflow().orElseThrow(PipelineMain::missingPubsubArgumentsException))
+                    .build();
+            var eventPublisher = new PubsubEventBuilder().newPublisher(arguments.pubsubProject().orElse(arguments.project()),
```

Comment:
maybe we can get rid of the eventContext here as a part since we have already deprecated it? or does something still rely on this